examples/stable_diffusion/evaluation.py

```diff
@@ -0,0 +1,267 @@
+    parser.add_argument("--model_id", default="CompVis/stable-diffusion-v1-4", type=str)
+    parser.add_argument(
+        "--guidance_scale",
+        default=7.5,
+        type=float,
+        help="Guidance scale as defined in Classifier-Free Diffusion Guidance",
+    )
+    parser.add_argument("--num_inference_steps", default=50, type=int, help="Number of steps in diffusion process")
+    parser.add_argument(
+        "--seed",
+        default=None,
+        type=int,
+        help="The seed to give to the generator to generate deterministic results.",
+    )
+    parser.add_argument("--data_dir", default="quantize_data", type=str)
+    parser.add_argument("--sub_dir", default="optimized", type=str, help="Sub directory to save the data for optimized model test")
+    parser.add_argument("--num_data", default=10, type=int)
+    parser.add_argument("--train_ratio", default=0.5, type=float)
+    parser.add_argument("--image_size", default=512, type=int, help="Width and height of the images to generate")
+    return parser.parse_args(raw_args)
+
+
+def main(raw_args=None):
+    args = parse_args(raw_args)
+    real_images = None
+    dataset = load_dataset("phiyodr/coco2017", streaming=True)
+    train_data = dataset['train']
+    prompts = [sanitize_path(example["captions"][0]) for i, example in enumerate(train_data) if i < args.num_data]
+    real_images = get_real_images(train_data, args.num_data)
+    
+    train_num = math.floor(len(prompts) * args.train_ratio)
+    clip_score_fn = partial(clip_score, model_name_or_path="openai/clip-vit-base-patch16")
+
+    script_dir = Path(".") #Path(__file__).resolve().parent
+    unoptimized_path: Path = script_dir / args.data_dir / 'unoptimized'
+    optimized_path: Path = script_dir / args.data_dir / args.sub_dir
+
+    unoptimized_model_dir = script_dir / "models" / "unoptimized" / args.model_id
+    optimized_dir_name = "optimized-qdq"
+    optimized_model_dir = script_dir / "models" / optimized_dir_name / args.model_id
+
+    model_dir = unoptimized_model_dir if args.save_data else optimized_model_dir
+    pipeline = OnnxStableDiffusionPipelineWithSave.from_pretrained(
+        model_dir, provider="CPUExecutionProvider"
+    )
+    pipeline.save_data_dir = None
+
+    if args.save_data:
+        os.makedirs(unoptimized_path, exist_ok=True)
+        for i, prompt in enumerate(prompts):
+            logger.info(prompt)
+            if i < train_num:
+                pipeline.save_data_dir = script_dir / args.data_dir / "data" / prompt
+                os.makedirs(pipeline.save_data_dir, exist_ok=True)
+            else:
+                pipeline.save_data_dir = None
+            run_inference(pipeline, args, prompt, unoptimized_path)
+        get_clip_scores(prompts, unoptimized_path, clip_score_fn)
+        get_fid_scores(prompts, unoptimized_path, real_images)
+        #get_hpsv2_scores(unoptimized_path / "hpsv2", partial(run_inference, pipeline, args, pathIsFile=True))
```

Comment:
Still need this?

Reply:
I am running it for some results, will update soon. This needs 800 images so commented out. I will add a switch for it

Reply:
updated result